profiling/src/lib.rs

```diff
@@ -336,6 +306,68 @@ pub struct RequestLocals {
     pub vm_interrupt_addr: *const AtomicBool,
 }
 
+/// take a sample every X bytes
+const ALLOCATION_PROFILING_INTERVAL: f32 = 1024.0 * 30.0;
+
+pub struct AllocationProfilingStats {
+    /// number of bytes until next sample collection
+    next_sample: i64,
+}
+
+impl AllocationProfilingStats {
+    pub fn new() -> AllocationProfilingStats {
+        AllocationProfilingStats {
+            next_sample: AllocationProfilingStats::next_sampling_interval(),
+        }
+    }
+
+    fn next_sampling_interval() -> i64 {
+        Poisson::new(ALLOCATION_PROFILING_INTERVAL)
+            .unwrap()
+            .sample(&mut rand::thread_rng()) as i64
+    }
+
+    pub fn track_allocation(&mut self, len: u64) {
+        self.next_sample -= len as i64;
+
+        if self.next_sample > 0 {
+            return;
+        }
+
+        let scale = 1.0 / (1.0 - (len as f64 * -1.0 / ALLOCATION_PROFILING_INTERVAL as f64).exp());
+
+        let count = 1.0 * scale;
+        let bytes = len as f64 * scale;
+
+        self.next_sample = AllocationProfilingStats::next_sampling_interval();
+
+        REQUEST_LOCALS.with(|cell| {
+            // Panic: there might already be a mutable reference to `REQUEST_LOCALS`
+            let locals = cell.try_borrow();
+            if locals.is_err() {
+                return;
+            }
+            let locals = locals.unwrap();
+
+            if !locals.profiling_experimental_allocations_enabled {
+                return;
+            }
+
```

Comment:
There might be no need for this check, as we register in `rinit()` (or not) and therefore there should not be a call to this function when allocation profiling is disabled

```suggestion
```